scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java

```diff
@@ -1224,6 +1224,27 @@ public Void doInTransaction(Session session) {
         });
     }
 
+    public void updateJobAndTasksStatuses(final InternalJob job, final Set<TaskId> taskIds, TaskStatus taskStatus) {
+        executeReadWriteTransaction((SessionWork<Void>) session -> {
+
+            updateTasksStatuses(taskIds, taskStatus, session);
+
+            JobInfo jobInfo = job.getJobInfo();
+
+            session.getNamedQuery("updateJobAndTasksState")
```

Comment:
maybe you can factorize this part of the code with the same block in updateJobAndTasksState